### PR TITLE
Use === everywhere

### PR DESCRIPTION
**Pull request by deepu105**
- Please make sure the below checklist is followed for Pull Requests.

- [ ] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
- [ ] Tests are added where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by pascalgrimaud** [comment]
Long to review it, but LGTM

File: generators/entity/templates/server/src/main/java/package/repository/_EntityRepository.java [hunk 1]
```diff
@@ -20,29 +20,29 @@
 
 import <%=packageName%>.domain.<%=entityClass%>;
 import org.springframework.stereotype.Repository;
-<% if (databaseType == 'cassandra') { %>
+<% if (databaseType === 'cassandra') { %>
 import com.datastax.driver.core.*;
 import com.datastax.driver.mapping.Mapper;
 import com.datastax.driver.mapping.MappingManager;<% } %><% if (databaseType=='sql') { %>
```
Comment (pascalgrimaud) [inline generators/entity/templates/server/src/main/java/package/repository/_EntityRepository.java]: it can be done in another PR, need to be changed to: ` === 'sql'`

File: generators/entity/templates/server/src/main/java/package/repository/_EntityRepository.java [hunk 2]
```diff
@@ -20,29 +20,29 @@
...
     for (r of relationships) {
-        if (r.relationshipType == 'many-to-one' && r.otherEntityName == 'user') {
+        if (r.relationshipType === 'many-to-one' && r.otherEntityName === 'user') {
             importList = true;
         }
     }
-    if (importList == true) {
+    if (importList === true) {
 _%>
 import java.util.List;<% }} %><% if (databaseType=='mongodb') { %>
```
Comment (pascalgrimaud) [inline generators/entity/templates/server/src/main/java/package/repository/_EntityRepository.java]: same here: ` === 'mongodb'`

File: generators/entity/templates/server/src/test/java/package/web/rest/_EntityResourceIntTest.java [hunk 3]
```diff
@@ -328,149 +328,149 @@ public void setup() {
...
         <%_ } _%>
 
         // An entity with an existing ID cannot be created, so this API call must fail
         rest<%= entityClass %>MockMvc.perform(post("/api/<%= entityApiUrl %>")
             .contentType(TestUtil.APPLICATION_JSON_UTF8)
-            .content(TestUtil.convertObjectToJsonBytes(<%= entityInstance %><% if (dto == 'mapstruct') { %>DTO<% } %>)))
+            .content(TestUtil.convertObjectToJsonBytes(<%= entityInstance %><% if (dto === 'mapstruct') { %>DTO<% } %>)))
             .andExpect(status().isBadRequest());
 
         // Validate the Alice in the database
```
Comment (pascalgrimaud) [inline generators/entity/templates/server/src/test/java/package/web/rest/_EntityResourceIntTest.java]: Why validating Alice here? cc @jdubois ? :)

File: generators/entity/templates/server/src/test/java/package/web/rest/_EntityResourceIntTest.java [hunk 4]
```diff
@@ -17,22 +17,22 @@
...
-<% if (authenticationType == 'uaa') { %>
+<% if (authenticationType === 'uaa') { %>
 import <%=packageName%>.config.SecurityBeanOverrideConfiguration;
 <% } %>
 import <%=packageName%>.domain.<%= entityClass %>;
 <%_ for (idx in relationships) { // import entities in required relationships
         const relationshipValidate = relationships[idx].relationshipValidate;
         const otherEntityNameCapitalized = relationships[idx].otherEntityNameCapitalized;
-        if (relationshipValidate != null && relationshipValidate === true) { _%>
+        if (relationshipValidate !== null && relationshipValidate === true) { _%>
```
Comment (pascalgrimaud) [inline generators/entity/templates/server/src/test/java/package/web/rest/_EntityResourceIntTest.java]: I think we can remove the `=== true`, but same comment: it should be done in another PR, it would be simplier

**Comment by deepu105** [comment]
@pascalgrimaud yes the remaining needs to be taken care of seperately. Would you like to PR the ones you found?

**Comment by pascalgrimaud** [comment]
yes i will do
